### PR TITLE
Add test section for OTLP Log ingestion in backend

### PR DESCRIPTION
**Pull request by IbraheemA**
## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
The lint fails. Just execute `./format.sh` and all should be good.

**Comment by cbeauchesne** [comment]
Changing to draft, you can set it back to normal once you need a review from us.

File: tests/otel_tracing_e2e/test_e2e.py [hunk 1]
```diff
@@ -168,6 +168,27 @@ def test_main(self):
             return
         validate_log_trace_correlation(otel_log_trace_attrs, trace_agent)
 
+
+        # The 2nd account has logs and traces sent via the backend OTLP intake endpoint
+        try:
+            log_agent = interfaces.backend.get_logs(
```
Comment (songy23) [inline tests/otel_tracing_e2e/test_e2e.py:174]:
```suggestion
            log_intake = interfaces.backend.get_logs(
```

Comment (songy23) [reply tests/otel_tracing_e2e/test_e2e.py:174]: Same for all reference below

File: tests/otel_tracing_e2e/test_e2e.py [hunk 2]
```diff
@@ -168,6 +168,27 @@ def test_main(self):
...
+                query=f"trace_id:{dd_trace_id}",
+                rid=rid,
+                dd_api_key=os.environ["DD_API_KEY_2"],
+                dd_app_key=os.environ["DD_APP_KEY_2"],
+            )
+            otel_log_trace_attrs = validate_log(log_agent, rid, "datadog_agent")
```
Comment (songy23) [inline tests/otel_tracing_e2e/test_e2e.py:180]: Replace `datadog_agent` with the log source in the endpoint

File: tests/otel_tracing_e2e/test_e2e.py [hunk 3]
```diff
@@ -168,6 +168,27 @@ def test_main(self):
...
+            trace_agent = interfaces.backend.assert_otlp_trace_exist(
```
Comment (songy23) [inline tests/otel_tracing_e2e/test_e2e.py:181]:
```suggestion
            trace_intake = interfaces.backend.assert_otlp_trace_exist(
```

File: tests/otel_tracing_e2e/test_e2e.py [hunk 4]
```diff
@@ -168,6 +168,27 @@ def test_main(self):
...
+                request=self.r,
+                dd_trace_id=dd_trace_id,
+                dd_api_key=os.environ["DD_API_KEY_2"],
+                dd_app_key=os.environ["DD_APP_KEY_2"],
+            )
+        except ValueError:
+            logger.warning("Backend does not provide logs")
+            return
+        validate_log_trace_correlation(otel_log_trace_attrs, trace_agent)
```
Comment (songy23) [inline tests/otel_tracing_e2e/test_e2e.py:190]:
```suggestion
        validate_log_trace_correlation(otel_log_trace_attrs, trace_intake)
```

File: utils/otel_validators/validator_log.py [hunk 5]
```diff
@@ -1,3 +1,5 @@
+import pprint
```
Comment (songy23) [inline utils/otel_validators/validator_log.py:1]: Where is this used?

**Review by songy23 — APPROVED** [review]

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
Just remove the pprint if not used

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**